### PR TITLE
fix: convert undefined rpc attributes to expected null type

### LINKED ISSUE #151: bug: `transaction.to` does not match type for contract creation transactions
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

0.1.1

### Current Behavior

Transaction objects returned by public client methods like `getBlock` and `getTransaction` do not always match their type. Specifically, `Transaction["to"]` is sometimes undefined (for contract creation transactions) despite being [typed as `Address | null`](https://github.com/wagmi-dev/viem/blob/main/src/types/transaction.ts#L69-L70). 

Verified on Alchemy ([explorer response here](https://dashboard.alchemy.com/composer?share=eJyVmF1PGzsQhv.LXnOBEk4auAvRyaHSKSCgvakq5PVOkhVee2V7G2jFf693ExZ7PDbpDUIzz.v6a9Yf_V3wLatlcXF6UkiwO6Wfhv8bsFtVXbMGiovC.f_4AfugmTSM21rJy5crZrbFG3fT9kFTXHz.XfxkoutFTPAtNC_9cGHMQb0GbZxKsBJElnk9IZ0e1BPIhRBqxyQH2gkxOadLJnom0aUQyfl8AcsqZlnGZ0RSPu_Tewcc6tamekWAlKepm04wC8PELrdMbshxkljO799n4F3fesbsnfGduGrazsId07V98dRh3FdUUHabR6sZh0uh_NNYdm.SBJD3uO6aEnTW5YAkfJZMCFo_ZBIqb_FosQ.4Hv3nxzhXnQxqIghjvuzHEQ0UZ7CKhwMbQxHX7xmfK4weohGtalky43_uQRjzYGzduCr6j_Hh_hmsWgNc1cYq7ZcWSmDNhplbXQcbSRCOeLCH.QAr3hOUhqxdIpnWkksZpVN6Ykch0ym9V5nLvt6yY0nAx3tnx5rECf_lqoiFGqIEvaqFBR3vlGQ6qf9fbZLiIUcoaU2KvtVKrWN8Hyb4O6UsvVxjhlDduw_GbWBhY9l7itAFl4RhxfoWFrL6LCt4js0_4I9qYV8Bf9MGUnzUCj19MZL3Geo1a7Mn8i6HjzXr88Yc5WQOc3KM48gSzl8lF3DUmifJD1w.WucMm3I_7CFjVxKmEXaMX2ofS4HYc_sa0u60Qw5jGPMNe3bHlupvUSuAW9DxAUoikU8ta7nByn0QsxJ2wxD2GxzSoCShTcnSiluQleuIV5opjySJXdsIxNNGEZGLVlZxJb6550t4xaOyWG1cC3dsR98REwDlkTfIqutNJOlDEfcieVweb1FMd65qjHV3SHKVcNZXDye17C.GlnheUdmE2qyUvtnJoHEyjfQroZTG18MwjhTXqwe6q34i1twzEVw4.GhMxyRJuVEtlRCAS4FMI.0wHTkHCkh5DK.fhHyfQ8r7ljXj4oTCMOXranOlRAX6Zk32mUyHet87UAYJX_N_L3kU7mUBaLsM41jRgrtFoptAGMeKn9GO4kd9WoNr0T2SiHFEqVDXKj3MbqAYgz5rgGm_zXyaCSDw6JrGvfZ.gSvFhbW6Lt3731.rBOB7DE.m.WvWE.rRmEbvWy8Ys2u8YQXhmHflGcF9LCZ1ap8n04QeWsH2N4fEmZXnUo7ZTkVE7OL_JvV_zlfuoJw_clGDtPHBSSQjrdmyKZYMsdcfJ0XLNGu_9Xj.22Rx_nw_qybV_YR.qqYw.8RnvCorztZsMp.OYFaeT2dnk1l1Op.MoZqU.8zOS8b42dnpuuRnc16uix_vfwBR45e1)), unsure about other RPC providers.

### Expected Behavior

Objects returned by public client methods should always match their type. Ideas:
1. Update the type to also allow undefined (`to?: Address | null`)
2. Convert undefined values present in the RPC to `null` (prefer this!)

### Steps To Reproduce

```ts
const transaction = await publicClient.getTransation({
  hash: "0x96d2d92c7d3e87c6cdbdcafa2836e6b936426d0828ed2b569baac440fbc48cbf",
})

console.log(transaction.to === undefined)
```

### Link to Minimal Reproducible Example (StackBlitz, CodeSandbox, GitHub repo etc.)

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by jxom**
Fixes #151

Some RPC providers deviate away from [EIP-1474](https://eips.ethereum.org/EIPS/eip-1474) types and provide `undefined` for attributes that should be `null`. This PR converts those types from the RPC to `null` before handing it to the consumer.

### CODE REVIEW

File: .changeset/dirty-meals-think.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"viem": patch
+---
+
+Formatted `undefined` values from RPC as `null` to confirm to EIP-1474.
```
Comment (tmm) [inline .changeset/dirty-meals-think.md:5]:
```suggestion
Formatted `undefined` values from RPC as `null` to conform to EIP-1474.
```

File: src/utils/formatters/block.ts [hunk 2]
```diff
@@ -24,6 +24,9 @@ export function formatBlock(block: Partial<RpcBlock>) {
     difficulty: block.difficulty ? BigInt(block.difficulty) : undefined,
     gasLimit: block.gasLimit ? BigInt(block.gasLimit) : undefined,
     gasUsed: block.gasUsed ? BigInt(block.gasUsed) : undefined,
+    hash: block.hash ? block.hash : null,
+    logsBloom: block.logsBloom ? block.logsBloom : null,
+    nonce: block.nonce ? block.nonce : null,
```
Comment (tmm) [inline src/utils/formatters/block.ts:29]: Does `??` work here instead of ternary?

Comment (jxom) [reply src/utils/formatters/block.ts:29]: It would, just did `? :` to keep consistent, but can change.

Comment (tmm) [reply src/utils/formatters/block.ts:29]: Good with either. I generally find `??` less verbose, but fine with keeping things consistent here.

**Pull request merged**